### PR TITLE
Update rolesd

### PR DESCRIPTION
**Pull request by ryanrath**
**NOTE:** These changes are to be made along side: 
  - https://github.com/ubccr/xdmod-xsede/pull/181
  - ccr-private PR as soon as I can get one up

**NOTE 2:** Also, I started these before the end of the year so they're pointed @ 8.6. I get that we may need to move them to 9.0. 

## Description
`configuration/roles.json` needed to be moved to `configuration/roles.d/appkernels.json` to be in line with our configuration file naming methods. 

## Motivation and Context
This was causing problems in the XSEDE install as there is already a `configuration/roles.json` and so this information was not being included.  

## Tests performed
Manual tests were performed on my dev port. 

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: configuration/roles.d/appkernels.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+          ],
+          "javascriptClass": "XDMoD.Module.AppKernels",
+          "javascriptReference": "CCR.xdmod.ui.appKernels",
+          "userManualSectionName": "App Kernels",
+          "tooltip": "Displays data referencing the reliability and performance of grid resources"
+        }
+      ]
+    }
+  }
+}
```
Comment (jtpalmer) [inline configuration/roles.d/appkernels.json]: Why did the file contents change?  The indentation and some of the ordering is different as well as the removal of the + prefixes.

Comment (ryanrath) [reply configuration/roles.d/appkernels.json]: - I'm honestly not sure why the `tooltip` / `userManualSectionName` properties were flipped, I can certainly put those back. 
- The `+` was removed as it doesn't do anything with the current configuration code. 
- And the spacing change was just flub on my part, I must have formatted it before saving. I can also change this back to 4 space min.

Comment (ryanrath) [reply configuration/roles.d/appkernels.json]: I just saw the `referencing` / `reflecting` change, not sure what was up with that but I've reverted it.

File: configuration/roles.d/appkernels.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+    "roles": {
+        "po": {
```
Comment (jtpalmer) [inline configuration/roles.d/appkernels.json]: There is no `po` role in Open XDMoD.

Comment (ryanrath) [reply configuration/roles.d/appkernels.json]: *facepalm* yes, these changes should be moved to ccr-private && then this file reverted to grant everybody access.

File: classes/Rest/Controllers/AppKernelControllerProvider.php [hunk 3]
```diff
@@ -1628,13 +1617,16 @@ public function getRawPerformanceMap(Request $request, Application $app)
...
-                'end_date' => $endDate,
-                'resource' => array('data' => $user->getResources()),
-                'appKer' => $appKernels,
-                'problemSize' => $problemSizes
-            ));
+            $options = array(
+                    'start_date' => $startDate,
+                    'end_date' => $endDate,
+                    'appKer' => $appKernels,
+                    'problemSize' => $problemSizes
```
Comment (jtpalmer) [inline classes/Rest/Controllers/AppKernelControllerProvider.php]: This indentation looks off.

File: classes/Rest/Controllers/AppKernelControllerProvider.php [hunk 4]
```diff
@@ -1628,13 +1617,16 @@ public function getRawPerformanceMap(Request $request, Application $app)
...
+            );
+            if (!$user->hasAcl(ROLE_ID_PROGRAM_OFFICER)) {
+                $options['resource'] = array('data' => $user->getResources());
+            }
```
Comment (jtpalmer) [inline classes/Rest/Controllers/AppKernelControllerProvider.php]: Why does this apply to everyone but program officers?  (I'm assuming that program officers should have access to all resources, but does `$user->getResources()` not return all resources for program officers?)

**Pull request closed without merging**